sentry-rails/lib/sentry/rails/tracing/active_record_subscriber.rb

```diff
@@ -31,6 +31,10 @@ def self.subscribe!
                             connection.pool.db_config.configuration_hash
                           elsif connection.pool.respond_to?(:spec)
                             connection.pool.spec.config
+                          # CockroachDB pool shows up as NullPool, but we can grab
```

Comment:
`ConnectionPool` should be assigned automatically during the connection establishing process, e.g.

https://github.com/rails/rails/blob/9c22f35440ab85718ebf48e26b8944032c737193/activerecord/lib/active_record/connection_adapters/pool_config.rb#L70

The fact that for Cockroach DB it remains `NullPool` makes me think that it may not exactly follow Rails' convention to set up its adapter. To be clear, I didn't find the code that directly contribute to this result so I may be wrong. But IMO it's the most likely case.

If that's the case, then it's the adapter gem that needs to be fixes, not the SDK.

Reply:
I _think_ you're right, and perhaps adding janky patches instead of fixing the adapter itself is not the way 👀

But, I'm not deeply experienced enough, maybe I should go debug how cockroachDB adapter instanciates the connection some more.

Reply:
I also think that `NullPool` here is a code smell, especially since the cockroach adapter is deriving from the postgres adapter, I'll read the source tomorrow and try to figure out if a patch upstream is better.